janus_server/src/aggregator.rs

```diff
@@ -241,9 +241,18 @@ impl<C: Clock> Aggregator<C> {
         req_bytes: &[u8],
         auth_token: Option<String>,
     ) -> Result<Vec<u8>, Error> {
-        // Parse task ID early to avoid parsing the entire message before performing authentication.
-        // This assumes that the task ID is at the start of the message content.
-        let task_id = TaskId::decode(&mut Cursor::new(req_bytes))?;
+        // Parse aggregation job ID early to avoid parsing the entire message before performing
+        // authentication. We need to know the task to perform authentication (since the auth token
+        // is a per-task configuration), and we need to know the aggregation job ID to look up the
+        // task.
+        let job_id = AggregationJobId::decode(&mut Cursor::new(req_bytes))?;
+        let task_id = self
+            .datastore
+            .run_tx(|tx| {
+                Box::pin(async move { tx.lookup_task_id_for_aggregation_job_id(job_id).await })
```

Comment:
Note: I considered e.g. `self.task_aggregator_for_aggregation_job_id(job_id)`. However:

1. If this ends up being a shim to support initial interop testing but we eventually include the `task_id` field, I don't want to spend too much time thinking about the caching logic.

2. If this ends up living forever, i.e. the WG doesn't include the `task_id` field or another indication of the task ID in aggregate continue requests, we might want to be careful about caching since aggregation jobs are pretty short-lived. E.g. for `prio3`, we have one AggregateInitializeReq & one AggregateContinueReq for each aggregation job, so we have one chance to get a cache hit -- if the AggregateContinueReq happens to go to a different process than the AggregateInitializeReq (pretty likely once we start scaling), simple in-memory caching will be ineffectual.

For that reason I decided just to look up the task ID for now.